### ISSUE TITLE
Files from stdlib/c_src not included in dist

### ISSUE THREAD

**Issue by nordlander** [actonlang/acton#278]
Stdlib C header files that include additional headers from stdlib/c_src no longer work when the including file is copied into dist/types. One example is stdlib/src/numpy.h, which in turn includes "../c_src/ndselect.h". The latter path is only valid inside stdlib/src, not from within dist/types.

Is the solution to copy stdlib/c_src into dist as well (and thereby reinterpret the meaning of "../c_src")? Or to expand the headers copied into dist/types so that any includes from c_src get inlined?

**Comment by nordlander** [comment]
An example that triggers the above error is the program examples/loess_smoother5.act (which now passes the type-checker again but chokes during C compilation).

**Comment by plajjan** [comment]
Ugh. Yeah, not sure. I felt everything was going so well with our stdlib-as-a-project restructuring up until I hit these includes and came up with this../c_src business.

We should discuss this. On Wednesday?

**Comment by plajjan** [comment]
I was pondering this a bit and at least I remember what I was going for. I mean, it's not super thought through, like I wrote previously, this was something I encountered while restructuring and I tried to solve it right there and then... anyhow, my thinking was that the public interface definition, like `types/numpy.h` should define the entire interface. Then the rest is internals.

In practice, I think this means that like `stdlib/src/numpy.h` should not include header files outside. `stdlib/src/numpy.h` should contain the entire public interface of the acton `numpy` module! If we need to access other things, like call a library, then we can include files in other directories from `stdlib/src/numpy.c`.

I'm no C expert, so maybe my thinking here is flawed, but this was what I was going for. I was thinking of how I wanted to implement the JSON module and keep the internals hidden etc.
